code/modules/mafia/controller.dm

```diff
@@ -592,10 +597,35 @@
 						continue
 					player.body.forceMove(get_turf(player.assigned_landmark))
 				if(failed.len)
-					to_chat(usr, "List of players who no longer had a body (if you see this, the game is runtiming anyway so just hit \"New Game\" to end it")
+					to_chat(usr, "List of players who no longer had a body (if you see this, the game is runtiming anyway so just hit \"New Game\" to end it)")
 					for(var/i in failed)
 						var/datum/mafia_role/fail = i
 						to_chat(usr, fail.player_key)
+			if("debug_setup")
+				var/list/debug_setup = list()
+				var/list/rolelist_dict = list()
+				var/done = FALSE
+				var/cancelled = FALSE
+				for(var/p in subtypesof(/datum/mafia_role))
+					var/datum/mafia_role/path = p
+					rolelist_dict[initial(path.name) + " ([uppertext(initial(path.team))])"] = path
+				rolelist_dict = list("CANCEL", "FINISH") + list("Assistant (TOWN)" = /datum/mafia_role) + rolelist_dict
```

Comment:
Why hardcode assistant? Couldn't you use typesof()?

Reply:
see above. I can typesof but lists in byond either dump all the assistants into psychologist or dump all the psychologists into assistants... or something else horrible.